### PR TITLE
Fix class field for ha-icon-next/prev

### PR DESCRIPTION
**Pull request by steverep**
## Proposed change

Fixes https://github.com/home-assistant/frontend/pull/18087#issuecomment-1747675417

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by KTibow** [comment]
Can you explain why you put it in willUpdate when it's a value that is always the same?

**Comment by steverep** [comment]
Because it won't trigger an update and produces the desired result.

**Comment by KTibow** [comment]
Why not in the constructor?

**Comment by steverep** [comment]
I don't see why that matters?  It's there before the first and any subsequent renders, whereas in the constructor it can be overridden by setting the icon externally.

File: src/components/ha-icon-next.ts [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
 import { customElement } from "lit/decorators";
 import { HaSvgIcon } from "./ha-svg-icon";
 
 @customElement("ha-icon-next")
 export class HaIconNext extends HaSvgIcon {
-  path = document.dir === "ltr" ? mdiChevronRight : mdiChevronLeft;
+  protected willUpdate(changedProperties: PropertyValues<this>) {
+    super.willUpdate(changedProperties);
+    this.path = document.dir === "ltr" ? mdiChevronRight : mdiChevronLeft;
+  }
```
Comment (bramkragten) [inline src/components/ha-icon-next.ts:11]:
```suggestion
  @property() public path =
    document.dir === "ltr" ? mdiChevronRight : mdiChevronLeft;
```

**Review by piitaya — APPROVED** [review]

**Pull request merged**